### PR TITLE
Add http method for getting application data (credit: Kailas, adapted by me).

### PR DESCRIPTION
**Pull request by luka-ethernal**

### CODE REVIEW

File: src/http.rs [hunk 1]
```diff
@@ -93,44 +122,83 @@ impl Service<Request<Body>> for Handler {
...
+									r#"{{"block": {}, "confidence": {}, "serialisedConfidence": {:?}}}"#,
+									block_num, confidence, serialised_conf
+								)
+								.to_owned(),
+							)
+						},
+						"appdata" => {
+							let block = match_app_url(path).unwrap();
+							let app_id = cfg.app_id.unwrap();
+							println!("fetching data for app_id: {}", app_id);
```
Comment (prabal-banerjee) [inline src/http.rs:163]: Can we turn this into `log`?

**Review by prabal-banerjee — APPROVED** [review]

File: src/http.rs (hunk 1 above)
Comment (luka-ethernal) [reply src/http.rs]: Done.

File: src/http.rs [hunk 2]
```diff
@@ -93,44 +122,83 @@ impl Service<Request<Body>> for Handler {
...
-						let mut not_found = Response::default();
-						*not_found.status_mut() = StatusCode::NOT_FOUND;
-						not_found.headers_mut().insert(
-							ACCESS_CONTROL_ALLOW_ORIGIN,
-							"*".parse().expect("* to be parsable"),
-						);
-						Ok(not_found)
+					let path = req.uri().path();
+					let v: Vec<&str> = path.split('/').collect();
+					match v[2] {
```
[the remark this answers is not in the document]
Comment (luka-ethernal) [reply src/http.rs]: Done.

**Review by sh3ll3x3c — APPROVED** [review]

File: src/http.rs [hunk 3]
```diff
@@ -29,161 +29,113 @@ pub fn serialised_confidence(block: u64, factor: f64) -> Option<String> {
...
-		},
-		Err(error) => {
-			log::error!("Bad HTTP server host:port ({host}:{port}) found in config file: {error}");
-		},
-	}
+	let host = cfg.http_server_host.clone();
+	let port = cfg.http_server_port;
+
+	let get_mode =
+		warp::path!("v1" / "mode").map(move || warp::reply::json(&Mode::from(cfg.app_id)));
```
Comment (kroos47) [inline src/http.rs:111]: thanks for changing it to warp 😄

**Review by kroos47 — APPROVED** [review]

**Review by prabal-banerjee — APPROVED** [review]

**Pull request merged**